contracts/GenericLender/GenericLenderBase.sol

```diff
@@ -34,19 +34,47 @@ abstract contract GenericLenderBase is IGenericLender {
     address public override strategy;
     IERC20 public want;
     string public override lenderName;
-
     uint256 public dust;
 
-    constructor(address _strategy, string memory name) public {
+    event Cloned(address indexed clone);
+
+    constructor(address _strategy, string memory _name) public {
+        _initialize(_strategy, _name);
+    }
+
+    function _initialize(address _strategy, string memory _name) internal {
+        require(address(strategy) == address(0), "Lender already initialized");
+
         strategy = _strategy;
         vault = VaultAPI(IBaseStrategy(strategy).vault());
         want = IERC20(vault.token());
-        lenderName = name;
+        lenderName = _name;
         dust = 10000;
 
         want.safeApprove(_strategy, uint256(-1));
     }
 
+    function initialize(address _strategy, string memory _name) external {
```

Comment:
i dont think we need this. and can make clone internal so that only the implementation has external callables